src/github.com/mozilla-services/pushgo/simplepush/app.go

```diff
@@ -294,7 +293,11 @@ func (a *Application) closeClients() {
 	}
 }
 
-func (a *Application) CloseOnce() error {
+func (a *Application) Close() error {
+	return a.closeOnce.Do(a.close)
+}
+
+func (a *Application) close() error {
```

Comment:
Kind of a nit, but we only ever call close() from Close(), and then we're doing some stack juggling to call the function. I like the pattern of Closer, but it seems like we could just call ``` if !a.closeOnce.IsClosed() { 
    ...  
    a.closeOnce.Close()
} ```
and do the same without the stack push/pops.